packages/grid/x-data-grid-pro/src/hooks/features/columnResize/gridColumnResizeApi.ts

```diff
@@ -0,0 +1,47 @@
+import { GridColDef } from '@mui/x-data-grid';
+
+export const DEFAULT_GRID_AUTOSIZE_OPTIONS = {
+  includeHeaders: false,
+  excludeOutliers: false,
+  outliersFactor: 1.5,
+  expand: false,
+};
+
+export type GridAutosizeOptions = {
+  /**
+   * The columns to autosize. By default, applies to all columns.
+   */
+  columns?: GridColDef[];
+  /**
+   * If true, include the header widths in the calculation
+   * @default false
+   */
+  includeHeaders?: boolean;
+  /**
+   * If true, width outliers will be ignored.
+   * @default false
+   */
+  excludeOutliers?: boolean;
```

Comment:
**Optional:** To me, excluding outliers is an amazing feature that should be expected by most of users. How about keeping the naming to match `includeHeaders` but having it turned on by default (by keeping the default value to `false`)?
```suggestion
  /**
   * If true, width outliers will be ignored.
   * @default false
   */
  includeOutliers?: boolean;
```

Reply:
Matching name: Not sure, I have a slight preference for `excludeOutliers` for some reason.

Default value: Yes, excluding outliers is an interesting behavior. I'd be ok with having it on by default. @cherniavskii Any opinion?

Reply:
I've included both these changes.

Reply:
Apologies for being late to the party, but I'd still like to share my opinion 🙃
While excluding outliers is a cool feature, I would expect autosizing to include outliers by default, because otherwise, it feels like autosizing doesn't work as expected (some of the values might still be truncated).

Would be great to have the `autosizeOptions` documented along with the default values 👍🏻

Reply:
One thing we overlooked is the experimental flag (or `unstable_` prefix) for the new feature so that we can iterate on default behavior and/or prop names 😅
cc @MBilalShafi @romgrk @joserodolfofreitas